[bug] Mishandling of Meson built-in options in MesonToolchain

### Describe the bug

I believe some built-in options are not handled correctly in the Meson Toolchain.

For example, the `b_staticpic` option is `True` by default in Meson. However, the template in `meson/toolchain.py` handles this value like this:

```jinja
    {% if b_staticpic %}
    b_staticpic = {{b_staticpic}}
    {% endif %}
```

So, users cannot set `b_staticpic` to `False` through Conan. I believe Conan codebase uses `None` to model when users do not specify a value. So most checks should be coded as:

```jinja
    {% if b_staticpic is not None %}
    b_staticpic = {{b_staticpic}}
    {% endif %}
```

Right?

### How to reproduce it

_No response_

> > b_ndebug is a boolean not a string, so it should be unquoted. If I force a value myself, then it gets the proper type. But, by default, it will be set to a string as shown by this code snippet. Looking at Meson, there is a third value that is indeed a string "if-release", but I do not believe Conan uses it.
> 
> That is not correct. If you define it as boolean `false` or `true` without quotes, Meson will warn you, telling it it is not a boolean and that you use a string `"false"` or `"true"`. This happens because the variable has to accomodate to the `"if-release"` which is not boolean, so the internal type of `b_ndebug` is string, not boolean.
> 
> I tried it in the PR linked above [#18843](https://github.com/conan-io/conan/pull/18843), you can see the message there.

I stand corrected

Regarding the linker script config, here is what I get

```sh
riscv64-unknown-elf-gcc   /**skipped**/ -Wl,--gc-sections -static '-T"/home/michael/.conan2/p/b/polarb941b78627c25/p/linker/mpfs-ddr-loaded-by-boot-loader.ld"'  /**skipped**/
/home/michael/.conan2/p/b/softc34418972624e6/p/bin/../lib/gcc/riscv64-unknown-elf/8.3.0/../../../../riscv64-unknown-elf/bin/ld: cannot open linker script file "/home/michael/.conan2/p/b/polarb941b78627c25/p/linker/mpfs-ddr-loaded-by-boot-loader.ld": No such file or directory
collect2: error: ld returned 1 exit status

```
It seems the double quotes are considered part of the path. Conan or Meson tries to escape them by enclosing that argument into single quotes. Since the double quotes are considered part of the path to the linker script, I get a linker error

Note that if I inject this linker script manually in the toolchain recipe package info, everything works

```python
        self.conf_info.append(
            "tools.build:exelinkflags",
            f"-T{os.path.join(self.package_folder, 'linker', 'mpfs-ddr-loaded-by-boot-loader.ld'}",
        )

```

--- a/conan/tools/meson/toolchain.py
+++ b/conan/tools/meson/toolchain.py
@@ -463,7 +463,7 @@ class MesonToolchain:
                                                 check_type=list)
         linker_scripts = self._conanfile_conf.get("tools.build:linker_scripts", default=[],
                                                   check_type=list)
-        linker_script_flags = ['-T"' + linker_script + '"' for linker_script in linker_scripts]
+        linker_script_flags = ['-T' + linker_script for linker_script in linker_scripts]
         defines = self._conanfile_conf.get("tools.build:defines", default=[], check_type=list)
         sys_root = [f"--sysroot={self._sys_root}"] if self._sys_root else [""]
         ld = (sharedlinkflags + exelinkflags + linker_script_flags + sys_root + self.extra_ldflags

--- a/test/functional/toolchains/cmake/test_cmake_toolchain.py
+++ b/test/functional/toolchains/cmake/test_cmake_toolchain.py
@@ -1442,3 +1442,49 @@ def test_cmaketoolchain_check_function_exists():
     c.run("build . ")  # Release breaks the check
     assert "Looking for pow - found" in c.out
     # And it no longer crashes
+
+
+@pytest.mark.skipif(platform.system() != "Linux", reason="Linker scripts in Linux only")
+def test_cmake_linker_scripts():
+    conanfile = textwrap.dedent("""
+       from conan import ConanFile
+       from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+
+       class Test(ConanFile):
+           exports_sources = "CMakeLists.txt", "src/*"
+           settings = "os", "compiler", "arch", "build_type"
+           generators = "CMakeToolchain"
+
+           def layout(self):
+               cmake_layout(self)
+
+           def build(self):
+               cmake = CMake(self)
+               cmake.configure()
+               cmake.build()
+       """)
+
+    main = "int main() {}"
+
+    cmakelists = textwrap.dedent("""
+        set(CMAKE_CXX_COMPILER_WORKS 1)
+        set(CMAKE_CXX_ABI_COMPILED 1)
+        cmake_minimum_required(VERSION 3.15)
+        project(Test CXX)
+        add_executable(example main.cpp)
+        """)
+    profile = textwrap.dedent("""
+       include(default)
+       [conf]
+       tools.build:linker_scripts=['{{profile_dir}}/mylinkscript.ld']
+       """)
+
+    c = TestClient()
+    c.save({"conanfile.py": conanfile,
+            "CMakeLists.txt": cmakelists,
+            "main.cpp": main,
+            "mylinkscript.ld": "",
+            "profile": profile})
+    c.run('build . -pr=profile', assert_error=True)
+    # This error means the linker script was fonud and loaded, it failed because empty
+    assert "PHDR segment not covered by LOAD segment" in c.out

--- a/test/functional/toolchains/meson/test_cross_compilation.py
+++ b/test/functional/toolchains/meson/test_cross_compilation.py
@@ -244,3 +244,49 @@ def test_use_meson_toolchain():
     # Build locally
     c.run('build . --profile:host=android')
     assert "conanfile.py (hello/0.1): Calling build()" in c.out
+
+
+@pytest.mark.tool("meson")
+@pytest.mark.tool("ninja")
+@pytest.mark.skipif(platform.system() != "Linux", reason="Linker scripts in Linux only")
+def test_linker_script():
+    # https://github.com/conan-io/conan/issues/18535
+    conanfile_py = textwrap.dedent("""
+    from conan import ConanFile
+    from conan.tools.meson import Meson
+    from conan.tools.files import load
+
+    class App(ConanFile):
+        settings = "os", "arch", "compiler", "build_type"
+        generators = "MesonToolchain"
+
+        def layout(self):
+            self.folders.build = "build"
+
+        def build(self):
+            meson = Meson(self)
+            try:
+                meson.configure()
+            finally:
+                errors = load(self, "meson-logs/meson-log.txt")
+                self.output.info(errors)
+    """)
+    meson_build = textwrap.dedent("""
+        project('tutorial', 'cpp')
+        executable('demo', 'main.cpp')
+        """)
+    main_cpp = gen_function_cpp(name="main")
+    profile = textwrap.dedent("""
+        include(default)
+        [conf]
+        tools.build:linker_scripts=['{{profile_dir}}/mylinkscript.ld']
+        """)
+    c = TestClient()
+    c.save({"conanfile.py": conanfile_py,
+            "meson.build": meson_build,
+            "main.cpp": main_cpp,
+            "mylinkscript.ld": "",
+            "profile": profile})
+    c.run('build . -pr=profile', assert_error=True)
+    # This error means the linker script was fonud and loaded, it failed because empty
+    assert "PHDR segment not covered by LOAD segment" in c.out

--- a/test/integration/toolchains/meson/test_mesontoolchain.py
+++ b/test/integration/toolchains/meson/test_mesontoolchain.py
@@ -230,7 +230,6 @@ def test_custom_arch_flag_via_toolchain():
     assert re.search(r"cpp_link_args =.+-mmy-flag.+", content)
 
 
-
 def test_linker_scripts_via_conf():
     profile = textwrap.dedent("""
         [settings]
@@ -256,8 +255,10 @@ def test_linker_scripts_via_conf():
 
     t.run("install . -pr:b=profile -pr=profile")
     content = t.load(MesonToolchain.native_filename)
-    assert "c_link_args = ['-flag0', '-other=val', '-m64', '-flag5', '-flag6', '-T\"/linker/scripts/flash.ld\"', '-T\"/linker/scripts/extra_data.ld\"']" in content
-    assert "cpp_link_args = ['-flag0', '-other=val', '-m64', '-flag5', '-flag6', '-T\"/linker/scripts/flash.ld\"', '-T\"/linker/scripts/extra_data.ld\"']" in content
+    assert ("c_link_args = ['-flag0', '-other=val', '-m64', '-flag5', '-flag6', "
+            "'-T/linker/scripts/flash.ld', '-T/linker/scripts/extra_data.ld']") in content
+    assert ("cpp_link_args = ['-flag0', '-other=val', '-m64', '-flag5', '-flag6', "
+            "'-T/linker/scripts/flash.ld', '-T/linker/scripts/extra_data.ld']") in content
 
 
 def test_correct_quotes():

